src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/UseSecureCookiesASPNetCore.cs

```diff
@@ -179,6 +179,8 @@ public override void Initialize(AnalysisContext context)
                                         InterproceduralAnalysisConfiguration.Create(
                                             compilationAnalysisContext.Options,
                                             SupportedDiagnostics,
+                                            rootOperationsNeedingAnalysis.First().Item1.Syntax.SyntaxTree,
```

Comment:
@dotpaul - For option value computed in CompilationEndAction, we do not have a specific tree/document to query the options. I have taken a simple route of just using the first tree for computing option value. You would need to switch to a non-CompilationEndAction based analysis approach if you'd like to support differing options between trees in a compilation for such analyzers.